### ISSUE TITLE
Layout/ExtraSpacing fails to catch specific edgecase

### ISSUE THREAD

**Issue by Kache** [rubocop/rubocop#8125]
I've reduced the repro case to a very small example below.

Highly-suspect is the use of "`end`" in the keyword-similar `ab_end`, of course.

Changing the alignment of text by changing variable lengths "avoids" this issue, for example the offense is properly caught if `ab_end` is renamed to `abc_end`.

--------

## Expected behavior

Should detect extra spaces between `select` and `do`.

## Actual behavior

It doesn't.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

[].select  do |x|
  x.ab_end == 1
end
```
`rubocop --force-default-config ./spacing_test.rb`

## RuboCop version

```sh
0.85.0 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.7.1 x86_64-darwin18)
```

**Comment by rrosenblum** [comment]
I was able to reproduce this issue. You are correct this appears to be related to `ab_end`. It seems to be an issue with the length of the statement rather than the usage of `end` in the name.

This produces the same edge case and no offense is created.
```ruby
[].select  do |x|
  x.ab_foo == 1
end
```

Part of the check for extra spacing will check to see if even if there is extra spacing on one line, that the code might be aligned with an element from the line before or the line after. In this case the `do` appears to be aligned with `==` on the next line so no offense is registered.

This might be tricky to fix because I assume any tweak to the check for previous and next line may result in breaking something else. Maybe we need to limit what kind of nodes we check multi-line alignment for.

**Comment by Kache** [comment]
I think I've got a good metric to use:

The Cop should be able to safely assume the user is attempting to align code only **iff** the lines are also indented the same.

**Comment by rrosenblum** [comment]
I really like the idea of trying to only check alignment if the indentation is the same. As you pointed out, there are always some edge cases to consider. 

I believe that @jonas054 worked on some of the original code for handling extra spaces and alignment. He might have some ideas here. I would also be interested in what some of the other contributors think about this @bbatsov, @marcandre, @koic, @Drenmi

**Comment by jonas054** [comment]
Yes, I did some work together with @jfelchner on `Layout/ExtraSpacing`. I agree that we should be able to get rid of some false negatives by ruling out some situations where extra spacing is clearly not potential alignment. But it's not as easy as requiring the lines to have the same indentation. Look at this _modified_ spec example for this cop:

```ruby
    it 'accepts aligned values of an implicit hash literal' do
      expect_no_offenses(<<~RUBY)
        register(street1: '1 Market',
                 state:   'CA')
      RUBY
    end
```

**Comment by jfelchner** [comment]
Most of what the extra spacing cop does now is already based on alignment.  Specifically around making sure that `=` are aligned.  I'm pretty sure that @bbatsov and I put in some pretty robust test cases, so as long as those all still pass, it should be ok, but that also doesn't mean there aren't missing test cases and something could break. ;)

**Comment by ydakuka** [comment]
No longer relevant. Everything is now working as expected.
